Customizing/global/plugins/Services/Cron/CronHook/UserOrguImport/classes/Factory.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+/**
+ * General plugin factory. Also creates sub-namespace-factories.
+ */
+
+namespace CaT\IliasUserOrguImport;
+
+use CaT\UserOrguImport as UOI;
+
+class Factory
+{
+	/**
+	 * @param	\ilDB	$db
+	 * @param	\ilRbacAdmin	$rbacadmin
+	 * @param	\ilRbacReview	$rbacreview
+	 * @param	\ilSetting	$settings
+	 * @param	\ilOrgUnitTree	$org_unit_tree
+	 * @param	\ilTree	$tree
+	 * @param	ErrorReporting\ErrorCollection  $error_collection
+	 * @param	\ilLog
+	 */
+	public function __construct(
+		\ilDB $db,
+		\ilRbacAdmin $rbacadmin,
+		\ilRbacReview $rbacreview,
+		\ilSetting $settings,
+		\ilObjOrgUnitTree $org_unit_tree,
+		\ilTree $tree,
+		\ilRepUtil $rep_utils,
+		ErrorReporting\ErrorCollection $error_collection,
+		\ilLog $illog
+	) {
+
+		$this->db = $db;
+		$this->rbacadmin = $rbacadmin;
+		$this->rbacreview = $rbacreview;
+		$this->settings = $settings;
+		$this->org_unit_tree = $org_unit_tree;
+		$this->tree = $tree;
+		$this->rep_utils = $rep_utils;
+		$this->error_collection = $error_collection;
+		$this->illog = $illog;
+	}
+
+	/**
+	 * Creates a factory for the User subnamespace.
+	 *
+	 * @return	User\DEVKUserFactory
+	 */
+	public function UserFactory()
+	{
+		return new User\UserFactory(
```

Comment:
These fat __constructors with a lot of dependencies make me suspicious, because they suggest to me, that the classes are doing to much things.